include/aws/http/request_response.h

```diff
@@ -28,11 +28,39 @@ struct aws_http_connection;
  */
 struct aws_http_stream;
 
+/**
+ * A lightweight HTTP header struct.
+ * Note that the underlying strings are not owned by the byte cursors.
+ */
 struct aws_http_header {
     struct aws_byte_cursor name;
     struct aws_byte_cursor value;
 };
 
+/**
+ * The definition for an HTTP request.
+ * This datastructure may be transformed (ex: signing the request) before it is used to create a stream.
+ *
+ * The request keeps internal copies of its trivial strings (method, path, headers)
+ * but does NOT take ownership of its body stream.
+ *
+ * Do not access the members of this struct directly, use the aws_http_request_XYZ() functions.
+ */
+struct aws_http_request {
```

Comment:
Why not just forward declare this and make it opaque?

Reply:
done